scene/main/viewport.cpp

```diff
@@ -3612,9 +3628,32 @@ void Viewport::_propagate_exit_world_3d(Node *p_node) {
 }
 
 void Viewport::set_use_xr(bool p_use_xr) {
-	use_xr = p_use_xr;
+	if (use_xr != p_use_xr) {
+		use_xr = p_use_xr;
+
+		RS::get_singleton()->viewport_set_use_xr(viewport, use_xr);
+
+		if (use_xr) {
+			// Force the viewport to resize right away, so that in NOTIFICATION_INTERNAL_PROCESS
+			// we can compare the XR interfaces size with the current size and view count, to avoid
+			// resizing every frame.
+			if (XRServer::get_singleton() != nullptr) {
+				Ref<XRInterface> xr_interface = XRServer::get_singleton()->get_primary_interface();
+				if (xr_interface.is_valid() && xr_interface->is_initialized()) {
+					Size2 xr_size = xr_interface->get_render_target_size();
+					_set_size((Size2i)xr_size, Size2i(), Rect2i(), Transform2D(), true, xr_interface->get_view_count(), true);
+				}
```

Comment:
again, if no valid xr_interface is found the viewport size should be set to 0,0.